### PR TITLE
[test/plugin] immigrate gateway scenario

### PR DESCRIPTION
**Pull request by zhaoyuguang**
Please answer these questions before submitting pull request

- Why submit this pull request?
- [ ] Bug fix
- [x] New feature provided
- [ ] Improve performance

- Related issues
#3583

### CODE REVIEW

**Comment by wu-sheng** [comment]
Check #3590. The number of each test case is required now. We haven't documented yet, sorry.

**Comment by wu-sheng** [comment]
FYI @dmsolr

**Comment by wu-sheng** [comment]
First don't add into group2, Jetty tests cost the longest time. Don't make our test slower :P

Then your own test case failure.

File: test/plugin/scenarios/gateway-scenario/gateway-dist/bin/startup.sh [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+home="$(cd "$(dirname $0)"; pwd)"
+
+java -jar ${agent_opts} "-Dskywalking.agent.service_name=gateway-projectA-scenario" ${home}/../libs/gateway-projectA-scenario.jar &
+sleep 1
+
+java -jar ${agent_opts} "-Dskywalking.agent.service_name=gateway-projectB-scenario"  ${home}/../libs/gateway-projectB-scenario.jar &
```
Comment (dmsolr) [inline test/plugin/scenarios/gateway-scenario/gateway-dist/bin/startup.sh]: `service_name` and `applications` of `expectedData.yml` don't match. :)

**Comment by zhaoyuguang** [comment]
> First don't add into group2, Jetty tests cost the longest time. Don't make our test slower :P
> 
> Then your own test case failure.

OK，already move to Group1

File: Jenkinsfile-Agent-Test [hunk 2]
```diff
@@ -93,7 +99,7 @@ pipeline {
                     stages {
                         stage('jetty 9.x (63)') {
                             steps {
-                                sh 'bash test/plugin/run.sh --build_id=${BUILD_ID} jetty-scenario'
+                                sh 'bash test/plugin/run.sh --build_id=test_201910121736 jetty-scenario'
```
Comment (arugal) [inline Jenkinsfile-Agent-Test]: Should use```${BUILD_ID}```,right?

Comment (zhaoyuguang) [reply Jenkinsfile-Agent-Test]: Thanks~

File: test/plugin/scenarios/gateway-scenario/gateway-dist/bin/startup.sh [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+home="$(cd "$(dirname $0)"; pwd)"
+
+cp -rf /usr/local/skywalking/scenario/agent/optional-plugins/apm-spring-cloud-gateway-2.x-plugin-*.jar /usr/local/skywalking/scenario/agent/plugins
```
Comment (dmsolr) [inline test/plugin/scenarios/gateway-scenario/gateway-dist/bin/startup.sh]: It doesn't recommend. I am sorry. Please remove it when `RunningMode` provided.

Comment (zhaoyuguang) [reply test/plugin/scenarios/gateway-scenario/gateway-dist/bin/startup.sh]: OK

Comment (wu-sheng) [reply test/plugin/scenarios/gateway-scenario/gateway-dist/bin/startup.sh]: @zhaoyuguang Do you miss this?

**Pull request merged**